src/uct/ib/base/ib_iface.c

```diff
@@ -286,7 +288,7 @@ void uct_ib_address_pack(const union ibv_gid *gid, uint16_t lid,
                 /* Site-local */
                 ib_addr->flags  |= UCT_IB_ADDRESS_FLAG_SUBNET16;
                 *(uint16_t*) ptr = gid->global.subnet_prefix >> 48;
-            } else {
+            } else if (gid->global.subnet_prefix != UCT_IB_LINK_LOCAL_PREFIX) {
```

Comment:
need to update `uct_ib_iface_address_size` then?

Reply:
yep, fixed